lib/rules/amd-function-arity.js

```diff
@@ -54,6 +83,16 @@ module.exports = function (context) {
                 expected: dependencyCount,
                 actual: actualParamCount
             });
+        } else if (dependencyNodes.length > callbackParams.length && allowedExtraDependencies && !allowExtraDependencies) {
```

Comment:
Though, I just noticed that the counts are already assigned above, so this could be even more concise: 
```js
dependencyCount > actualParamCount && Array.isArray(allowExtraDependencies)
```